### DISCUSSION TITLE
PollingBlockTracker error in CI pipeline

### DISCUSSION THREAD

**Discussion by Babitabisht** [ConsenSys-archive/truffle#5404] [General Q&A]
Hi,

I have a CI pipeline, which is running truffle tests in each commit. The pipeline is running ganache before the truffle test.
The issue is - 
 ganache is running fine on the defined port but the truffle test fails intermittently, with the following error -

![Screenshot from 2022-08-05 16-36-17](https://user-images.githubusercontent.com/40245628/183067840-256141d1-3abf-446b-8d0f-d61a031d75cf.png)

Not sure, what the cause is.  I have already tried different solutions available on the internet like increasing networkCheckTimeout and timeoutBlocks but the issue is the same.

This is what my network provider in truffle-config.js looks like -
    development: {
      host: "127.0.0.1",
      port: 9546,
      network_id: "*",
      networkCheckTimeout: 20000,
      timeoutBlocks: 300
    },

**Comment by apdan510** [comment]
@Babitabisht 

Thanks for reaching out! Can you please share the full exact command you are using to run "Truffle test" Could also share a code snippet where it runs ganache before truffle test? 

Thank you!

**Reply by Babitabisht** [reply]
Thanks, But already got the solution.   Running ganache as a separate service ( using docker  )  in the pipeline worked out!!

**Reply by apdan510** [reply]
great!
